asf_tools/composite.py

```diff
@@ -165,82 +170,74 @@ def make_composite(outfile, infiles=None, path=None, pol=None, resolution=None):
 
-    logging.info(f"Full extent of mosaic is {ulx,uly} to {lrx,lry}")
+    # Get extent of union of all images
+    full_ul, full_lr, full_trans, full_proj = get_full_extent(raster_info)
 
-    x_pixels = abs(int((ulx - lrx) / pixel_size_x))
-    y_pixels = abs(int((lry - uly) / pixel_size_y))
+    nx = int(abs(full_ul[0] - full_lr[0]) // resolution)
+    ny = int(abs(full_ul[1] - full_lr[1]) // resolution)
 
-    logging.info(f"Output size is {x_pixels} samples by {y_pixels} lines")
+    outputs = np.zeros((ny, nx))
+    weights = np.zeros(outputs.shape)
+    counts = np.zeros(outputs.shape, dtype=np.int8)
 
-    outputs = np.zeros((y_pixels, x_pixels))
-    weights = np.zeros((y_pixels, x_pixels))
-    counts = np.zeros((y_pixels, x_pixels), dtype=np.int8)
     logging.info("Calculating output values")
-
-    for fi, x_max, x_min, y_max, y_min in extents:
-        if pol in fi:
-            logging.info(f"Processing file {fi}")
-            logging.info(f"File covers {x_max,y_min} to {x_min,y_max}")
-
-            logging.info("Reading areas")
-            x_size, y_size, trans, proj, areas = saa.read_gdal_file(saa.open_gdal_file(fi.replace(f"_{pol}_reproj",
-                                                                                                  "_area_reproj")))
-
-            logging.info("Reading values")
-            x_size, y_size, trans, proj, values = saa.read_gdal_file(saa.open_gdal_file(fi))
-
-            out_loc_x = (x_max - ulx) / pixel_size_x
-            out_loc_y = (y_min - uly) / pixel_size_y
-            end_loc_x = out_loc_x + x_size
-            end_loc_y = out_loc_y + y_size
-
-            logging.info(f"Placing values in output grid at {int(out_loc_x)}:{int(end_loc_x)} "
-                         f"and {int(out_loc_y)}:{int(end_loc_y)}")
-
-            temp = 1.0/areas
-            temp[values == 0] = 0
-            mask = np.ones((y_size, x_size), dtype=np.uint8)
-            mask[values == 0] = 0
-
-            outputs[int(out_loc_y):int(end_loc_y), int(out_loc_x):int(end_loc_x)] += values * temp
-            weights[int(out_loc_y):int(end_loc_y), int(out_loc_x):int(end_loc_x)] += temp
-            counts[int(out_loc_y):int(end_loc_y), int(out_loc_x):int(end_loc_x)] += mask
+    for raster, info in raster_info.items():
+        logging.info(f"Processing raster {raster}")
+        logging.info(f"Raster upper left: {info['cornerCoordinates']['upperLeft']}; "
+                     f"lower right: {info['cornerCoordinates']['lowerRight']}")
+
+        logging.info(f"Reading raster values {raster}")
+        rds = gdal.Open(raster)
+        values = rds.GetRasterBand(1).ReadAsArray()
+        del rds  # How to close w/ gdal
+
+        raster_split = raster.split('_')
+        raster_pol = raster_split[-1].split('.')[0]
+        area_suffix = raster_split[-1].replace(raster_pol, 'area')
+        area_raster = '_'.join(raster_split[:-1] + [area_suffix])
```

Comment:
This might be more defensive than necessary and I still don't know if I like it better than:

```python
        if '_reproj' in raster:
            suffix = '_reproj'
        else:
            suffix = ''	
        area_raster = '_'.join(raster.replace('_reproj', '').split('_')[:-1] + [f'area{suffix}.tif'])
```
*Note: I changed reprojected file naming from `_reproj.tif` to `.reproj.tif` so splitting on `_` doesn't add an extra element*

Maybe:
```python
        raster_split = raster.split('_')
        area_suffix = '.'.join(['area'] + raster_split[-1].split('.')[1:])
        area_raster = '_'.join(raster_split[:-1] + [area_suffix])
```
ehh...